src/todo/UI.lua

```diff
@@ -145,13 +145,20 @@ function todo.create_add_edit_frame(player)
     })
 
 
-    local players, _ = todo.get_player_list()
-    table.add({
+    local players, _, c = todo.get_player_list()
+    -- The count is off by 1 so if the result is 2 we get only one player
+    local t = {
         type = "drop-down",
         name = "todo_add_assignee_drop_down",
-        items = players,
-        selected_index = 1
-    })
+        items = players
+    }
+    if( todo.is_auto_assign(player) and c == 1 ) then
+        t.selected_index = 2
+    else
+        t.selected_index = 1
+    end
+    table.add(t)
```

Comment:
What I meant is more like this
```
local index = 0
if (.) then
index = 1
else
index = 2
end

table.add({
...
selected_index = index
})
````

Then you don't have to mix execution and details that much.

Reply:
Then something like:
```lua
local index = 2
if( . ) then
  index = 1
end
table.add({
....
selected_index = index
})
```